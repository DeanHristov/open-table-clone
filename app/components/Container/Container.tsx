import { FC, ReactNode } from "react";
import classNames from "classnames";

export interface IContainerProps {
  children: ReactNode;
  width?: string;
}

const Container: FC<IContainerProps> = ({ children, width = "w-auto" }) => {
  const defaultStyles = [
    `bg-white w-full ${width} rounded p-3`,
    "shadow border border-gray-200",
  ];
  return <div className={classNames(defaultStyles)}>{children}</div>;
};

export default Container;
