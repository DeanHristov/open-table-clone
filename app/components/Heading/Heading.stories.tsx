import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Heading, { IHeadingProps } from "./index";

export default {
  title: "Components/Heading",
  component: Heading,
} as Meta;

export const Default: Story<IHeadingProps> = (args) => <Heading {...args} />;

Default.args = {
  title: "Heading title",
  secondary: false,
};
