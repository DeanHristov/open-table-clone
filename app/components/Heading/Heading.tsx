"use client";

import { FC } from "react";
import classNames from "classnames";

export interface IHeadingProps {
  title: string;
  secondary?: boolean;
  large?: boolean;
}

const Heading: FC<IHeadingProps> = ({
  title,
  large = false,
  secondary = false,
}) => {
  const defaultStyles = ["text-neutral-8", "font-bold", "leading-6"];
  const styles = classNames(defaultStyles, {
    "text-2xl": !large,
    "text-6xl": large,
  });

  return (
    <div className={`pb-4 ${secondary ? "border-b" : ""} border-gray-300`}>
      <h1 className={styles}>{title}</h1>
    </div>
  );
};

export default Heading;
