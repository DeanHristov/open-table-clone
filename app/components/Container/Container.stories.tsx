import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Container, { IContainerProps } from "./index";
import Heading from "../Heading";

export default {
  title: "Components/Container",
  component: Container,
} as Meta;

export const Restaurant: Story<IContainerProps> = (args) => (
  <Container {...args}>
    <Heading large title={"Milesstone Grill"} />
    <div className="mt-4">
      <Heading secondary title={"X photos"} />
    </div>
  </Container>
);

Restaurant.args = {
  width: "md:w-[70%]",
};
