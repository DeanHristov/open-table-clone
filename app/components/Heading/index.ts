import Heading, { IHeadingProps } from "./Heading";

export type { IHeadingProps };
export default Heading;
