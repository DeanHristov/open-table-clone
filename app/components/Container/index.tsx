import Container, { IContainerProps } from "./Container";

export type { IContainerProps };

export default Container;
