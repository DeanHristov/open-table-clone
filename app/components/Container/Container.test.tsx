import { render } from "@testing-library/react";
import Heading from "../Heading";
import Container from "./index";

describe("Component: <Container {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(
      <Container>
        <Heading title={"Just a test"} />
      </Container>
    );

    expect(container).toMatchSnapshot();
  });

  it("Should be able to render with dynamic width", () => {
    const { container } = render(
      <Container width={"w-[70%]"}>
        <Heading title={"Just a test"} />
      </Container>
    );

    expect(container).toMatchSnapshot();
  });
});
