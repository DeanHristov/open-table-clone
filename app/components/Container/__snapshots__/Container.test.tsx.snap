// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Container {...} /> Should be able to render default state 1`] = `
<div>
  <div
    class="bg-white w-full w-auto rounded p-3 shadow border border-gray-200"
  >
    <div
      class="pb-4  border-gray-300"
    >
      <h1
        class="text-neutral-8 font-bold leading-6 text-2xl"
      >
        Just a test
      </h1>
    </div>
  </div>
</div>
`;

exports[`Component: <Container {...} /> Should be able to render with dynamic width 1`] = `
<div>
  <div
    class="bg-white w-full w-[70%] rounded p-3 shadow border border-gray-200"
  >
    <div
      class="pb-4  border-gray-300"
    >
      <h1
        class="text-neutral-8 font-bold leading-6 text-2xl"
      >
        Just a test
      </h1>
    </div>
  </div>
</div>
`;
